merkle/smt/writer.go

```diff
@@ -32,7 +32,7 @@ type NodeBatchAccessor interface {
 	// The returned hashes may be missing or be nil for empty subtrees.
 	Get(ctx context.Context, ids []tree.NodeID2) (map[tree.NodeID2][]byte, error)
 	// Set applies the given node hash updates.
-	Set(ctx context.Context, upd []NodeUpdate) error
+	Set(ctx context.Context, upd []Node) error
```

Comment:
This `upd` and the one below in `Split` is a possible for rename but not essential.